packages/blocks/src/lib/navigation-map/components/input/geometry.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang='ts'>
+
+import { Radio } from '@viamrobotics/prime-core'
+import { createEventDispatcher } from 'svelte';
+import type { Geometry, Shapes } from '$lib';
+import VectorInput from '../vector-input.svelte';
+import { createGeometry } from '../../lib/geometry';
+
+export let geometry: Geometry;
+
+const dispatch = createEventDispatcher<{ input: Geometry }>();
+
+const handleShapeSelect = (event: CustomEvent<{ value: string }>) => {
+  const nextType = event.detail.value.toLowerCase() as Shapes;
+  dispatch('input', createGeometry(nextType));
+};
+
+const handleDimensionsInput = (event: CustomEvent<number[]>) => {
+  const [x, y, z] = event.detail;
+
+  switch (geometry.type) {
+    case 'box': {
+      if (x) {
+        geometry.length = x;
+      }
+      if (y) {
+        geometry.width = y;
+      }
+      if (z) {
+        geometry.height = z;
+      }
+      break;
+    }
+    case 'sphere': {
+      if (x) {
+        geometry.radius = x;
+      }
+      break;
+    }
+    case 'capsule': {
+      if (x) {
+        geometry.radius = x;
+      }
+      if (y) {
+        geometry.length = y;
+      }
+      break;
+    }
+  }
+
+  dispatch('input', geometry);
```

Comment:
- Is this going to miss changes when `x === 0`, `y === 0`, or `z === 0`?
- Given that we `dispatch` the updated object, we should not be mutating a passed-in prop like this. It could pull the rug out from under the parent component or another sibling that received the same `Geometry` object

```suggestion
  const [x, y, z] = event.detail;
  let nextGeometry: Geometry
  
  switch (geometry.type) {
    case 'box': {
      nextGeometry = {
        ...geometry,
        length: x ?? geometry.length,
        width: y ?? geometry.width,
        height: z ?? geometry.height,
      }
      break;
    }
    case 'capsule': {
      nextGeometry = {
        ...geometry, 
        radius: x ?? geometry.radius,
        length: y ?? geometry.length,
      }
      break;
    }
    case 'sphere': {
      nextGeometry = {
        ...geometry,
        radius: x ?? geometry.radius,
      }
      break;
    }
  }

  dispatch('input', nextGeometry);
```